test/PowerShellEditorServices.Test/Language/SemanticTokenTest.cs

```diff
@@ -0,0 +1,190 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Management.Automation.Language;
+using System.Threading.Tasks;
+using Microsoft.PowerShell.EditorServices.Services.TextDocument;
+using Microsoft.PowerShell.EditorServices.Handlers;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document.Proposals;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models.Proposals;
+using Xunit;
+
+namespace Microsoft.PowerShell.EditorServices.Test.Language
+{
+    public class SemanticTokenTest
+    {
+        [Fact]
+        public async Task TokenizesFunctionElements()
+        {
+            string text = @"
+function Get-Sum {
+    param( [int]$a, [int]$b )
+    return $a + $b
+}
+";
+            ScriptFile scriptFile = new ScriptFile(
+                // Use any absolute path. Even if it doesn't exist.
+                DocumentUri.FromFileSystemPath(Path.Combine(Path.GetTempPath(), "TestFile.ps1")),
+                text,
+                Version.Parse("5.0"));
+
+            foreach (Token t in scriptFile.ScriptTokens)
+            {
+                List<SemanticToken> mappedTokens = new List<SemanticToken>(PsesSemanticTokensHandler.ConvertToSemanticTokens(t));
+                switch (t.Text)
+                {
+                    case "function":
+                    case "param":
+                    case "return":
+                        Assert.Collection(mappedTokens, sToken => Assert.Equal(SemanticTokenType.Keyword, mappedTokens[0].Type));
```

Comment:
You're using `Assert.Collection` here, but you never use `sToken` and you always use index zero for `mappedTokens`. What you can do with `Assert.Collection` is:

```csharp
Assert.Collection(mappedTokens,
    // "function"
    sToken => Assert.Equal(SemanticTokenType.Foo, sToken.Type,
    // "Get-Sum"
    sToken => Assert.Equal(SemanticTokenType.Bar, sToken.Type,
    // "param"
    sToken => Assert.Equal(SemanticTokenType.Baz, sToken.Type,
    // ...
    sToken => Assert.Equal(SemanticTokenType.asdf, sToken.Type,
    sToken => Assert.Equal(SemanticTokenType.fdsas, sToken.Type,
    sToken => Assert.Equal(SemanticTokenType.Keyw, sToken.Type,
    sToken => Assert.Equal(SemanticTokenType.Ke, sToken.Type,
)
```

where each additional lambda parameter is that index in the `List`.

It also checks that the count is correct as well.

This means you can get rid of the switch entirely.

Reply:
Oops the mappedTokens[0] was a remnant from an old test case. 
Also, mappedToken are the tokens mapped from the PS AST tokens (not all the tokens in the file)

Reply:
Missed that! Thanks!